### PR TITLE
Weather@mockturtl Update 2.1.2

### PR DESCRIPTION
**Pull request by Gr3q**
Don't know how to do the versioning anymore.

### Changelog:
**3.6**
* OpenWeatherMap using FOSS key and proper daily forecasts.

**3.8**
* Removed "require"-s, swapped them to imports
* Fixed bug when Cinnamon froze when the taskbar was manipulated when the applet was enabled on it.
* DarkSky was Getting the Forecast day names from sunrise time, fixed
* Humidity is rounded now.
* Location element is a button again, opens Data services webpages with more weather, or it can trigger a refresh if there was an error (there is no need anymore, but still).

@jaszhix Please check the readme on the Compatibility section, is that right?
I don't quite get it how the versioning works.
Basically from Cinnamon v3.2-3.6 the 3.6 folder is used, and Cinnamon v3.8< the 3.8 folder is used?

Right now 3.6 could support all the way back to 1.8.

### CODE REVIEW

**Comment by feren** [comment]
I don't know if it is my inner noob at getting the applets straight from the GitHub, or if this is an actual issue, but I tried getting the Pull Requested version from https://github.com/Gr3q/cinnamon-spices-applets/tree/rebase-dev so that I could confirm if this is fixed or not, and I've put the contents of the 'files' folder in ~/.local/share/cinnamon/applets, and the applet fails to load for me on the version present there, with this error logged in Looking Glass:
```
error t=2019-02-02T21:48:42Z 
[weather@mockturtl]: module is not defined
[weather@mockturtl]: Error importing applet.js from weather@mockturtl
```

Apologies if I did something wrong with installing the newer version and that's why, but I've had that issue every time I've tried it on the version present on the Pull Request Source GitHub...
Cinnamon 4.0.9

**Comment by Odyseus** [comment]
Hello, everybody.

@feren: Here is how you can checkout a PR into a new branch (at least it's how I do it):

```shell
$ git fetch origin pull/pr-number/head:weather_applet_test && git checkout weather_applet_test
```

Change `pr-number` for this PR number (2288) and `weather_applet_test` (the name of the branch) for whatever you want.

**Comment by feren** [comment]
Tried that and obtained the applet, and then copied the applet folder over to ~/.local/share/cinnamon/applets and unfortunately the same error is still happening.

I've got to go now, so good luck and hopefully this'll be fixed if it isn't just happening due to my noob-iness at installing these from GitHub Repositories.

**Comment by Gr3q** [comment]
@feren I thought I tested it properly, but I did not.

Moving the applet to different folder broke my imports.

So this:
image
does not work anymore in 3.8/applet.js

I will need to figure it out how to traverse folders with that method, or go back to this:
image


If you change those lines to the second method, the applet will load

**Comment by Gr3q** [comment]
So hold off with the pull request because the 3.8 version does not work atm what I'll fix tomorrow,

And I need to test that 3.6 loads in when it needs to

**Comment by Odyseus** [comment]
@Gr3q: Since the applet is **multiversion** now, you could just use **require** for the applet stored inside the 3.8 folder.

Also, only one **metadata.json** file should exist at the root of the applet directory and nowhere else. The Cinnamon documentation has a section called *Supporting multiple versions* where this is explained.

**Comment by jaszhix** [comment]
Ah, that would be my fault. I accidentally left metadata.json inside the 3.6 folder.

**Comment by Gr3q** [comment]
Both FIxed.

@Odyseus I could use require, but I wanted to make sure @MaxMotovilov to have one less problem extending compatibility of the new version, as he wants to do that

**Comment by cranes-bill** [comment]
@Gr3q:
>I still need to VM an older version of Mint to be 100% sure it works alright now

I tested the 778014f commit of your **rebase-dev** branch and can confirm that "**3.6**" applet works perfect on Cinnamon 3.6.7

**Comment by Gr3q** [comment]
@cranes-bill Thanks

**Comment by feren** [comment]
I can confirm this version now functions and fixes https://github.com/linuxmint/cinnamon-spices-applets/issues/2287 in Cinnamon 4.0.
